flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/ScalarFunctionsTest.scala

```diff
@@ -2661,6 +2665,37 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
     //      "1996-11-10 06:55:44.333")
   }
 
+  @Test
+  def testRandAndIf(): Unit = {
```

Comment:
Currently we only have tests for rand with seed, actually the case reported in FLINK-33541 is without seed. I know currently the test suite cannot do this (check a expression/query runs without checking the expected result, or checking the result is in a range), so I won't have a strong opinion that we must add such a test case.

Reply:
Let me take a look and will push again.